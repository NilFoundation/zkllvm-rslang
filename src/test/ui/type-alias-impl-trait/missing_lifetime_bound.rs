#![feature(type_alias_impl_trait)]

type Opaque<'a, T> = impl Sized;
fn defining<'a, T>(x: &'a i32) -> Opaque<T> { x }
//~^ ERROR: hidden type for `Opaque<'a, T>` captures lifetime that does not appear in bounds

fn main() {}
