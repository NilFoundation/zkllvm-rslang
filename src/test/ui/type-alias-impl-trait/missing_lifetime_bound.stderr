error[E0700]: hidden type for `Opaque<'a, T>` captures lifetime that does not appear in bounds
  --> $DIR/missing_lifetime_bound.rs:4:47
   |
LL | fn defining<'a, T>(x: &'a i32) -> Opaque<T> { x }
   |             --                                ^
   |             |
   |             hidden type `&'a i32` captures the lifetime `'a` as defined here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0700`.
